input/tar2/tar_input.go

```diff
@@ -75,6 +75,9 @@ func (i Input) Apply(destinationRoot string) <-chan error {
 			bucket := &fshash.MemoryBucket{}
 			Extract(tarReader, destinationRoot, bucket, i.hasherFactory)
 
+			// bucket processing may have created a root node if missing.  if so, we need to apply its props.
+			fs.PlaceFile(destinationRoot, bucket.Root().Metadata, reader)
```

Comment:
More because `bucket.Root().Metadata` is a dir, but yeah, actually, that reader could/should be nil for explicitness and clarity.